general.py

```diff
@@ -180,19 +180,131 @@ def run_bin_analysis(err_prop = False, indiv = False):
     
+    #Read in composite data tables
+    Mbin_valid_tbl = asc.read(path_init + fitspathM + filename_dict['bin_valid'])
+    Mbininfo_tbl = asc.read(path_init + fitspathM + filename_dict['bin_info'])
     
+    MLHbbin_valid_tbl = asc.read(path_init + fitspathMLHb + filename_dict['bin_valid'])
+    MLHbbininfo_tbl = asc.read(path_init + fitspathMLHb + filename_dict['bin_info'])
+    if revised_files == True:
+        Mbin_derivedprops_tbl = asc.read(path_init + fitspathM + filename_dict['bin_derived_prop_rev'])
+        MLHbbin_derivedprops_tbl = asc.read(path_init + fitspathMLHb + filename_dict['bin_derived_prop_rev'])
+    else:    
+        Mbin_derivedprops_tbl = asc.read(path_init + fitspathM + filename_dict['bin_derived_prop'])
+        MLHbbin_derivedprops_tbl = asc.read(path_init + fitspathMLHb + filename_dict['bin_derived_prop'])
+
+
+    #Read in individual data
+    Mbin_indiv_logM = Mbin_indiv_props_tbl[indv_names0[3]].data
+    MLHbbin_indiv_logM = MLHbbin_indiv_props_tbl[indv_names0[3]].data
+    MLHbbin_indiv_logLHb = MLHbbin_indiv_props_tbl[indv_names0[4]].data
+    
+    Mbin_indiv_HBETA = Mbin_indiv_props_tbl['HBETA_Flux_Observed'].data
+    Mbin_indiv_OII = Mbin_indiv_props_tbl['OII_3727_Flux_Observed'].data
+    Mbin_indiv_OIII5007 = Mbin_indiv_props_tbl['OIII_5007_Flux_Observed'].data
+    
+    MLHbbin_indiv_HBETA = MLHbbin_indiv_props_tbl['HBETA_Flux_Observed'].data
+    MLHbbin_indiv_OII = MLHbbin_indiv_props_tbl['OII_3727_Flux_Observed'].data
+    MLHbbin_indiv_OIII5007 = MLHbbin_indiv_props_tbl['OIII_5007_Flux_Observed'].data
+    
+    Mbin_indiv_metal = Mbin_indiv_derivedprops_tbl[temp_metal_names0[1]].data
+    Mbin_indiv_bin_detect = Mbin_indiv_bininfo_tbl[bin_names0[2]].data
+    
+    MLHbbin_indiv_metal = MLHbbin_indiv_derivedprops_tbl[temp_metal_names0[1]].data
+    MLHbbin_indiv_bin_detect = MLHbbin_indiv_bininfo_tbl[bin_names0[2]].data
+    
+    
+    #Read in composite data
+    Mbin_logM = Mbininfo_tbl[bin_mzevolve_names0[2]].data      
+    Mbin_metal = Mbin_derivedprops_tbl[temp_metal_names0[1]].data
+    Mbin_detect_col = Mbin_valid_tbl[bin_names0[2]].data
+    
+    MLHbbin_logM = MLHbbininfo_tbl[bin_mzevolve_names0[2]].data      
+    MLHbbin_metal = MLHbbin_derivedprops_tbl[temp_metal_names0[1]].data
+    MLHbbin_detect_col = MLHbbin_valid_tbl[bin_names0[2]].data
+    
+    
+    #Define detection and non-detection (with reliable 5007) arrays for bins
+    Mbin_detect = np.where(Mbin_detect_col == 1.0)[0]
+    Mbin_nondetect = np.where(Mbin_detect_col == 0.5)[0]
+    
+    MLHbbin_detect = np.where(MLHbbin_detect_col == 1.0)[0]
+    MLHbbin_nondetect = np.where(MLHbbin_detect_col == 0.5)[0]
+    
+    #Define detection and non-detection (with reliable 5007) arrays for individual galaxies
+    Mbin_indiv_detect = np.where((Mbin_indiv_bin_detect == 1.0) & (np.isfinite(Mbin_indiv_OIII5007) == True) &
```

Comment:
Consider using set logic with `&` and have separate where statements.  This would make it much easier to understand the sample selections.